File: samples/CslaNet/cs/DeepData/DeepData.Library/LineItemInfo.cs
using Csla;
using Csla.Data;
using System;

namespace DeepData.Library
{
	///<summary>
	///</summary>
	[Serializable]
	public partial class LineItemInfo : ReadOnlyBase<LineItemInfo>
	{
		#region " Business Methods "

		private int _orderId;
		///<summary>
		///</summary>
		public int OrderId
		{
			[global::System.Runtime.CompilerServices.MethodImpl(global::System.Runtime.CompilerServices.MethodImplOptions.NoInlining)]
			get
			{
				CanReadProperty("OrderId", true);
				return _orderId;
			}
		}

		private int _id;
		///<summary>
		///</summary>
		public int Id
		{
			[global::System.Runtime.CompilerServices.MethodImpl(global::System.Runtime.CompilerServices.MethodImplOptions.NoInlining)]
			get
			{
				CanReadProperty("Id", true);
				return _id;
			}
		}

		private string _product = "";
		///<summary>
		///</summary>
		public string Product
		{
			[global::System.Runtime.CompilerServices.MethodImpl(global::System.Runtime.CompilerServices.MethodImplOptions.NoInlining)]
			get
			{
				CanReadProperty("Product", true);
				return _product;
			}
		}

		private LineItemDetailList _detailList = LineItemDetailList.NewList();
		///<summary>
		///</summary>
		public LineItemDetailList Details
		{
			get
			{
				return _detailList;
			}
		}


		///<summary>
		///
		///            Override this method to return a unique identifying
		///            value for this object.
		///            
		///</summary>
		///
		protected override object GetIdValue()
		{
			return string.Format("{0}::{1}", _orderId, _id);
		}

		#endregion

		#region " Factory Methods "

		internal static LineItemInfo GetItem(SafeDataReader data)
		{
			return new LineItemInfo(data);

		}

		private LineItemInfo()
		{

		}

		private LineItemInfo(SafeDataReader data)
		{

			Fetch(data);

		}

		#endregion

		#region " Data Access "

		private void Fetch(SafeDataReader data)
		{
			_orderId = data.GetInt32("OrderId");
			_id = data.GetInt32("Id");
			_product = data.GetString("Product");

		}

		internal void LoadDetail(SafeDataReader data)
		{
			_detailList.LoadChild(data);

		}

		#endregion

	}
}

File: Samples/CslaNet/vb/DeepData/DeepData.Library/LineItemInfo.vb
<Serializable()> _
Public Class LineItemInfo
  Inherits ReadOnlyBase(Of LineItemInfo)

#Region " Business Methods "


  Private Shared OrderIdProperty As PropertyInfo(Of Integer) = RegisterProperty(Of Integer)(Function(c) c.OrderId)
  ''' <Summary>
  ''' Gets and sets the NewProperty value.
  ''' </Summary>
  Public Property OrderId() As Integer
    Get
      Return GetProperty(OrderIdProperty)
    End Get
    Private Set(ByVal value As Integer)
      LoadProperty(OrderIdProperty, value)
    End Set
  End Property


  Private Shared IdProperty As PropertyInfo(Of Integer) = RegisterProperty(Of Integer)(Function(c) c.Id)
  ''' <Summary>
  ''' Gets the NewProperty value.
  ''' </Summary>
  Public Property Id() As Integer
    Get
      Return GetProperty(IdProperty)
    End Get
    Private Set(ByVal value As Integer)
      LoadProperty(IdProperty, value)
    End Set
  End Property



  Private Shared ProductProperty As PropertyInfo(Of String) = RegisterProperty(Of String)(Function(c) c.Product)
  ''' <Summary>
  ''' Gets the NewProperty value.
  ''' </Summary>
  Public Property Product() As String
    Get
      Return GetProperty(ProductProperty)
    End Get
    Private Set(ByVal value As String)
      LoadProperty(ProductProperty, value)
    End Set
  End Property


  Private Shared DetailsProperty As PropertyInfo(Of LineItemDetailList) = RegisterProperty(Of LineItemDetailList)(Function(c) c.Details)
  ''' <Summary>
  ''' Gets the NewProperty value.
  ''' </Summary>
  Public ReadOnly Property Details() As LineItemDetailList
    Get
      If (Not FieldManager.FieldExists(DetailsProperty)) Then
        LoadProperty(DetailsProperty, LineItemDetailList.NewList)
      End If

      Return GetProperty(DetailsProperty)
    End Get
  End Property


  Protected Overrides Function GetIdValue() As Object
    Return String.Format("{0}::{1}", OrderId, Id)
  End Function

#End Region

#Region " Factory Methods "

  Friend Shared Function GetItem(ByVal data As SafeDataReader) As LineItemInfo

    Return New LineItemInfo(data)

  End Function

  Private Sub New()

  End Sub

  Private Sub New(ByVal data As SafeDataReader)

    Fetch(data)

  End Sub

#End Region

#Region " Data Access "

  Private Sub Fetch(ByVal data As SafeDataReader)

    LoadProperty(OrderIdProperty, data.GetInt32("OrderId"))
    LoadProperty(IdProperty, data.GetInt32("Id"))
    LoadProperty(ProductProperty, data.GetString("Product"))

  End Sub

  Friend Sub LoadDetail(ByVal data As SafeDataReader)

    Details.LoadChild(data)

  End Sub

#End Region

End Class
